src/nvim/ex_docmd.c

```diff
@@ -6948,10 +6948,39 @@ void free_cd_dir(void)
 
 #endif
 
+void apply_autocmd_dirchanged(char_u *new_dir, CdScope scope)
+{
+  dict_T *dict = get_vim_var_dict(VV_EVENT);
+  char buf[8];
+
+  switch (scope) {
+  case kCdScopeGlobal:
+    snprintf(buf, sizeof buf, "global");
+    break;
+  case kCdScopeTab:
+    snprintf(buf, sizeof buf, "tab");
+    break;
+  case kCdScopeWindow:
+    snprintf(buf, sizeof buf, "window");
+    break;
+  default:
```

Comment:
I added it because clang gave me this warning:

```
warning: enumeration value 'kCdScopeInvalid' not handled in switch [-Wswitch]
  switch (scope) {
          ^
```